File: backend/src/main/java/tn/esprit/forum/entities/Review.java
package tn.esprit.forum.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Review {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Setter(AccessLevel.NONE)
    Long idReview;
    Integer rating;
    String comment;
    @ManyToOne // na7ina l cascade bch ki nfaskhou l review ma yetfasa5ch l user
    User user;

    @ManyToOne
    Offre offre;
}

File: frontend/src/app/core/models/Review.ts
import { Offre } from "./Offre";
import { User } from "./user";

export class Review {
    idReview: number;
    rating: number = 0;   
    comment: string;
    user: User; // Assuming you have a User class in your Angular application
    offre: Offre;
  
  }
  
 
  